extensions/elation/actions/addVitals/addVitals.ts

```diff
@@ -0,0 +1,89 @@
+import { z } from 'zod'
+import { isNil, omitBy, isEmpty } from 'lodash'
+import { type Action, Category, validate } from '@awell-health/extensions-core'
+import { SettingsValidationSchema, type settings } from '../../settings'
+import { makeAPIClient } from '../../client'
+import type {
+  measurementInputSchema,
+  AddVitalsInputSchema,
+} from '../../types/vitals'
+import {
+  FieldsValidationSchema,
+  fields as elationFields,
+  dataPoints,
+} from './config'
+
+// Helper function to create measurement objects with optional notes
+const createMeasurement = (
+  value?: string,
+  note?: string
+): Array<z.infer<typeof measurementInputSchema>> | undefined => {
+  if (isNil(value)) {
```

Comment:
From experience, an `isEmpty` check on the value is safer.

**Why?**
Once you configured an action field with a static value and you remove it, the value is `""` and not `undefined`.